monix-reactive/shared/src/main/scala/monix/reactive/internal/builders/Zip3Observable.scala

```diff
@@ -56,7 +56,9 @@ private[reactive] final class Zip3Observable[A1, A2, A3, +R](
     // MUST BE synchronized by `lock`
     var continueP = Promise[Ack]()
     // MUST BE synchronized by `lock`
-    var completeWithNext = false
+    var sourcesCompleted: Int = 0
```

Comment:
Might be faster with N booleans or `Byte` but I'll just leave an issue about it.
It might not matter so I'd rather not spend time on it right now